### PR TITLE
extractor scripts and db changes

### PR DESCRIPTION
**Pull request by shaibrown5**

### CODE REVIEW

File: superx/information_extractors/item_info_extractor.py [hunk 1]
```diff
@@ -124,23 +124,16 @@ def extract_information_from_parsed_xml(self, xml_info_child_node):
...
-            db.session.commit()
-            # adding current_branch_price to the db and commit
-            current_branch_price = BranchPrice(item_code=item_code, price=price, update_date=update_date)
+            # if item is not in db then add it
+            if not bool(Product.query.filter_by(id=item_code).first()):
+                current_product = Product(id=item_code, name=item_name, quantity=quantity, is_weighted=is_weighted,
+                                          unit_of_measure=unit_of_measure)
+                db.session.add(current_product)
+
+            current_branch_price = BranchPrice(branch_id='', item_code=item_code, price=price, update_date=update_date)
```
Comment (shaibrown5) [inline superx/information_extractors/item_info_extractor.py]: I still need to add a part that gets the branch id, for now i wrote it as empty

Comment (simzacks) [reply superx/information_extractors/item_info_extractor.py]: We can merge it like this, but it won't work until you get the branch id.

Comment (shaibrown5) [reply superx/information_extractors/item_info_extractor.py]: will update the pr today with branch id

File: superx/information_extractors/item_info_extractor.py [hunk 2]
```diff
@@ -148,24 +130,19 @@ def extract_information_from_parsed_xml(self, xml_info_child_node):
...
+                current_product = Product(id=item_code, name=item_name, quantity=quantity, is_weighted=is_weighted,
+                                          unit_of_measure=unit_of_measure)
+                db.session.add(current_product)
+
+            branch_price_list.append(BranchPrice(branch_id=branch_id, item_code=item_code, price=price,
+                                                 update_date=update_date))
+
+        db.session.commit()
+        db.session.add_all(branch_price_list)
+        db.session.commit()
```
Comment (shaibrown5) [inline superx/information_extractors/item_info_extractor.py:148]: 2 separate commits are necessary as branch_price has a foreign key from the products table, and so firt all the products must be committed before the branch _prices are committed

Comment (simzacks) [reply superx/information_extractors/item_info_extractor.py:148]: That must be a limitation of the ORM.
When working directly with the database, as soon as you insert (before commit), the record gets the id and you can use it within the same transaction.
According to StackOverflow you can overcome this with:
https://stackoverflow.com/questions/28242523/get-inserted-key-before-commit-session

But if you want to implement it this way, we merge that.

Comment (shaibrown5) [reply superx/information_extractors/item_info_extractor.py:148]: i will try it out, the list object is very costly

**Review by simzacks — COMMENTED** [review]
LGTM. Are you ready to merge this?

**Pull request merged**